src/ray/util/event.cc

```diff
@@ -312,6 +367,33 @@ void RayEvent::SendMessage(const std::string &message) {
     event.mutable_custom_fields()->insert(mp.begin(), mp.end());
 
     EventManager::Instance().Publish(event, custom_fields_);
+  } else if (is_export_event) {
+    std::string event_id_buffer = std::string(kEventIDSize, ' ');
+    FillRandom(&event_id_buffer);
+    event_id = StringToHex(event_id_buffer);
+
+    rpc::ExportEvent export_event;
+    export_event.set_event_id(event_id);
+
+    std::string source_type_as_str = Event_SourceType_Name(context.GetSourceType());
+    rpc::ExportEvent_SourceType source_type_ele;
+    if (rpc::ExportEvent_SourceType_Parse(source_type_as_str, &source_type_ele)) {
+      export_event.set_source_type(source_type_ele);
+    } else {
+      throw std::invalid_argument("Invalid source_type type: " + source_type_as_str);
+    }
+    export_event.set_timestamp(current_sys_time_s());
+
+    std::string event_data_type_name = export_event_data_ptr_->GetTypeName();
+    if (event_data_type_name == "ray.rpc.ExportTaskEventData") {
```

Comment:
Can we use this kind of API instead of comparing string like this? https://github.com/ray-project/ray/blob/7b1731810f5159604756b29527e3159dc9448f43/src/ray/pubsub/publisher.cc#L347

Reply:
`export_event_data_ptr_` is a pointer to a generic `google::protobuf::Message` so it doesn’t seem to have a similar API. `inner_message_case` can be used in that example because `PubMessage.inner_message` is a `one_of` field. 

This function is finding the type of `export_event_data_ptr_` so it can populate the `one_of` field of `ExportEvent`. From what I found, `GetTypeName` seems to be the recommended way to get the type of `google::protobuf::Message`.